Functions/Quantification/xASL_quant_FSL.m

```diff
@@ -95,8 +95,54 @@
     end
 
     %% 4. Create option_file that contains options which are passed to the FSL command
-	% FSLOptions is a character array containing CLI args for the BASIL/FABBER command
-	FSLOptions = xASL_sub_FSLOptions(pathFSLOptions, x, bUseFabber, PWI, pathFSLInput, pathFSLOutput);
+	bMergingTxt = 0; % only active if SessionMergingList + last session of list -> merge FSLOptions.txt
+    SessionsToMerge = {}; % To be used later during the txt merging
+    % For testing:
+    % x.modules.asl.SessionMergingList = {{'ASL_0','ASL_1', 'ASL_2'}; {'ASL_3', 'ASL_4','ASL_5'}};
+   
+    % 4.1 Check if necessary to merge sessions (and FSLOptions.txt)
+    if isfield(x.modules.asl,'SessionMergingList')
+       index=0;
+            for V=1:numel(x.modules.asl.SessionMergingList)
+                if ismember(x.SESSION, x.modules.asl.SessionMergingList{V})
+                    index=V;
+                    break;
+                end
+            end
+            CurrentSessionList = x.modules.asl.SessionMergingList{index};
+
+            if strcmp(x.SESSION,CurrentSessionList{end}) % If the SESSION is the last of the list, we want to merge the .txt and concatenate it with the previous sessions of the list
+
+                bMergingTxt = 1; % only active for the last session of a Merging list -> merge FSLOptions.txt
+                bUseFabber = 1;
+                
+                % Find PWI4Ds of the sessions to merge
+                CurrentSession = x.SESSION;
+                ListSessions = CurrentSessionList(1:end-1);
+                CurrentScanPWIdir = sprintf('%s/PWI4D_FSLInput.nii',CurrentSession);
+
+                for nSession = 1: numel(ListSessions)
+
```

Comment:
We can do this path replacement nicer using internal parameters in x. I will do that later.

Reply:
Done